File: tests/workflow/consumer/rust/src/consumer/broadcasts/groupa_structa.rs
use super::{protocol, Consumer, Context};
use clibri::client;

pub async fn handler<E: client::Error>(
    event: protocol::GroupA::StructA,
    context: &mut Context,
    consumer: Consumer<E>,
) {
    context.broadcast.groupa_structa += 1;
    context.broadcast.check();
}

File: experimental/workflow_mt/producer/typescript/src/producer/beacons/groupa.structa.ts
import {
    Context,
    Producer,
    Identification,
    Filter,
    Protocol,
} from "../implementation/beacons";

export function emit(
    beacon: Protocol.GroupA.StructA,
    consumer: Identification,
    filter: Filter,
    context: Context,
    producer: Producer
): Promise<void> {
    return Promise.reject(
        new Error(`Handler for Protocol.GroupA.StructA isn't implemented.`)
    );
}